bot-components/GitHub_subscriptions.mli

```diff
@@ -0,0 +1,5 @@
+val receive_github :
+     secret:string
+  -> Cohttp.Header.t
+  -> string
+  -> (bool * GitHub_types.msg, string) result
```

Comment:
The one type I would keep in this module instead of moving it elsewhere is the `msg` type. It is not really a type that may need to be shared between `GitHub_subscriptions`, `GitHub_queries` and `GitHub_mutations`.